central/role/service/service_impl.go

```diff
@@ -230,7 +230,7 @@ func (s *serviceImpl) PostPermissionSet(ctx context.Context, permissionSet *stor
 }
 
 func (s *serviceImpl) PutPermissionSet(ctx context.Context, permissionSet *storage.PermissionSet) (*v1.Empty, error) {
-	err := s.roleDataStore.UpdatePermissionSet(ctx, permissionSet)
+	err := s.roleDataStore.UpdatePermissionSet(ctx, permissionSet, storage.Traits_IMPERATIVE)
```

Comment:
We should add tests on the service level to verify that update / delete is handled correctly, this way we add more than just the manual tests.

Reply:
What do you mean by Update/Delete handled correctly? Looking at the logic within `service_impl`, I don't think there is much to test(especially if we go with determining client source from context)